### ISSUE TITLE
损失函数

### ISSUE THREAD

**Issue by houyuchao** [yhcc/CNN_Nested_NER#33]
我想将二元交叉熵变换为多元交叉熵函数，将F.binary_cross_entropy更换为F.cross_entropy，不对预测分数和target矩阵做flat_scores = scores.reshape(-1)，flat_matrix = matrix.reshape(-1)直接使用scores 和matrix作为F.cross_entropy，但是在最后 loss = ((flat_loss.view(input_ids.size(0), -1)*mask).sum(dim=-1)).mean()这一步的时候出现了维度不匹配问题，经过交叉熵出来的损失函数维度变成了[8，15，5]而二元交叉熵的输出维度为[8，15，15，5]导致使用交叉熵后与mask维度不匹配，这该怎么办啊？
![image](https://github.com/user-attachments/assets/1da4697b-605b-48ab-818e-4ab78f532148)
作者有没有什么办法啊，求助求助

**Comment by houyuchao** [comment]
> 我想将二元交叉熵变换为多元交叉熵函数，将F.binary_cross_entropy更换为F.cross_entropy，不对预测分数和target矩阵做flat_scores = scores.reshape(-1)，flat_matrix = matrix.reshape(-1)直接使用scores 和matrix作为F.cross_entropy，但是在最后 loss = ((flat_loss.view(input_ids.size(0), -1)*mask).sum(dim=-1)).mean()这一步的时候出现了维度不匹配问题，经过交叉熵出来的损失函数维度变成了[8，15，5]而二元交叉熵的输出维度为[8，15，15，5]导致使用交叉熵后与mask维度不匹配，这该怎么办啊？ ![image](https://private-user-images.githubusercontent.com/162104094/352061841-1da4697b-605b-48ab-818e-4ab78f532148.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MDE3MTIsIm5iZiI6MTcyMTkwMTQxMiwicGF0aCI6Ii8xNjIxMDQwOTQvMzUyMDYxODQxLTFkYTQ2OTdiLTYwNWItNDhhYi04MThlLTRhYjc4ZjUzMjE0OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNVQwOTU2NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OTg4NzVmZGUzNGYwMDJlMTc1Y2M4MTRiOGE0ZTQyOWM2OGQzZDVlZTE1ZDNmNDJjMDkxYzEzYTZiMmUyYjhhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.m_d7oCPdXGRSBGD_q5nnMTcyBf7whxSQaX9ZQEhCMOw) 作者有没有什么办法啊，求助求助

还是说不可以替换损失函数呢？
